kernel/src/actions/mod.rs

```diff
@@ -100,10 +254,12 @@ pub struct Protocol {
     pub min_writer_version: i32,
     /// A collection of features that a client must implement in order to correctly
     /// read this table (exist only when minReaderVersion is set to 3)
-    pub reader_features: Option<Vec<String>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub reader_features: Option<HashSet<String>>,
     /// A collection of features that a client must implement in order to correctly
     /// write this table (exist only when minWriterVersion is set to 7)
-    pub writer_features: Option<Vec<String>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub writer_features: Option<HashSet<String>>,
```

Comment:
Since probably not too critical I was on the fence about changing this, but then I though since we will mostly be checking if a feature exists, might as well and hear what others have to say.

Reply:
I'd vote for leaving it as a `Vec`. Generally for  small numbers of items in the collection, a scan of the vec is faster.

Some data [here](https://gist.github.com/daboross/976978d8200caf86e02acb6805961195) although the test is slightly flawed. But for low numbers of elements `Vec` should generally be better.